app/state.go

```diff
@@ -491,18 +491,18 @@ func (uc *updateCommitState) Handle(ctx *StateContext, c Controller) (State, boo
 	// to roll back, so the rest (if any) will be committed in the next
 	// state.
 	installers := c.GetInstallers()
-	if len(installers) < 1 {
-		return uc.HandleError(ctx, c, NewTransientError(
-			errors.New("GetInstallers() returned empty list? Should not happen")))
-	}
-	err = installers[0].CommitUpdate()
-	if err != nil {
-		// we need to perform roll-back here; one scenario is when
-		// u-boot fw utils won't work after update; at this point
-		// without rolling-back it won't be possible to perform new
-		// update
-		merr := NewTransientError(errors.Errorf("update commit failed: %s", err.Error()))
-		return uc.HandleError(ctx, c, merr)
+	if len(installers) == 0 {
+		log.Warn("GetInstallers() returned empty list: Artifact with no payload")
```

Comment:
This was a key discovery: a bootstrap Artifact has 1 _installer_ but 0 _updateStorers_. From `mender` client point of view, we name variables as installers when we really mean storers :man_shrugging:.

So yes, this check is to detect a bootstrap Artifact and do not commit it (it has no _updateStorer_ that knows how to commit it).

Reply:
Ok, another example of the excellent (not!) API we have in mender-artifact. 🙂

Still, I am a little bit confused. You say "do not commit it", but yet the code continues after the log statement. And don't we need to commit, even for a bootstrap artifact? Maybe this is better discussed on a call, because I'm probably missing something here...

Reply:
Discussed on a call, the "commit" of the meta-data is done later on calling `CommitArtifactData`. Here in this `if` condition what we avoid is to "commit" the payload.

Reply:
Discussed f2f, will change log level, but otherwise this is good.